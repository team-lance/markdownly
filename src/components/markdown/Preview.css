.preview {
  padding-top: 1.125em;
}
