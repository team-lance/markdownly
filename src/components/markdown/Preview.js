import React from 'react';
import PropTypes from 'prop-types';
import marked from 'marked';
import styles from 'github-markdown-css/github-markdown.css';
import myStyles from './Preview.css';

function Preview({ markdown }) {
  const __html = marked(markdown);
  return <div className={`${styles['markdown-body']} ${myStyles.preview}`} dangerouslySetInnerHTML={{ __html }}></div>;
}

Preview.propTypes = {
  markdown: PropTypes.string.isRequired
};

export default Preview;
