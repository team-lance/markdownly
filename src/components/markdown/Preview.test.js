import React from 'react';
import { shallow } from 'enzyme';
import Preview from './Preview';

describe('Preview', () => {
  it('matches a snapshot', () => {
    const markdown = '## Some markdown text\n\n### Some description text';
    const wrapper = shallow(
      <Preview markdown={markdown}/>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
