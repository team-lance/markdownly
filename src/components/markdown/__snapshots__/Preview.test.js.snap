// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview matches a snapshot 1`] = `
<div
  className="markdown-body preview"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<h2 id=\\"some-markdown-text\\">Some markdown text</h2>
<h3 id=\\"some-description-text\\">Some description text</h3>
",
    }
  }
/>
`;
